api/auth.py

```diff
@@ -0,0 +1,24 @@
+import os
+
+from flask_login import current_user
+from models.role import Role
+from models.user import User
+
+def get_user(user_id) -> User:
+    return User('name', user_id)
+
+def requires_role(role: Role):
+    def decorator(func):
+        def wrapper(*args, **kwargs):
+            user = get_user(current_user.get_id())
+            if(authorisation_disabled() or user.role is role):
+                return func(*args, **kwargs)
+            else:
+                return 'Not Authorized'
```

Comment:
It would be nice to combine this with an appropriate status code, currently this shows the relevant text to the user but also responds & is logged as a 200. Flask makes that easy - `return 'Not Authorized', 401` would be a quick solution for that